### ISSUE TITLE
When I distribute my project, whether I do not need the uWSGI? 

### ISSUE THREAD

**Issue by moonshineBoy** [eventlet/eventlet#489]
I use the `eventlet.wsgi.server` in my project `wsgi.py`:

```
import eventlet.wsgi
eventlet.wsgi.server(eventlet.listen(('', 8000)), application)
```

Now I want to distribute my project.

When I use `uwsgi --ini uwsgi.ini`:

there will get bellow error:

```
mapped 208320 bytes (203 KB) for 4 cores
*** Operational MODE: threaded ***
django.setup()
Traceback (most recent call last):
  File "myProject/wsgi.py", line 30, in <module>
    eventlet.wsgi.server(eventlet.listen(('', 8000)), application)
  File "/Users/lll/Desktop/myProject/Project/myProject/venv/lib/python3.5/site-packages/eventlet/convenience.py", line 78, in listen
    sock.bind(addr)
OSError: [Errno 48] Address already in use
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 11413)
spawned uWSGI worker 1 (pid: 11418, cores: 4)
spawned uWSGI http 1 (pid: 11419)
```

and I use `lsof -i:8000` ferret out the `uwsgi` is using the `8000` port.

```
$ lsof -i:8000
COMMAND   PID USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
uwsgi   11413  ldl    4u  IPv4 0x9bf2de3fe3ee0367      0t0  TCP *:irdmi (LISTEN)
uwsgi   11419  ldl    4u  IPv4 0x9bf2de3fe3ee0367      0t0  TCP *:irdmi (LISTEN)
```
so the eventlet.wsgi can not listen the address: `eventlet.listen(('', 8000))`

Whether when I distribute my project, I do not need the uWSGI ?  use the eventlet.wsgi.server directly?

**Comment by temoto** [comment]
TL;DR: if possible, distribute without WSGI server. If you must talk HTTP directly, use uwsgi.

This seems like two distinct issues.

* of course you must not bind two programs to same socket address, this is not specific to eventlet/uwsgi; choose one WSGI server to listen on unique (free) address
* whatever you do, **do not open eventlet.wsgi to public internet**

It was good intention of Python community to "distribute" a `def app(environ, start_response):`, your deployment person or infrastructure ought to provide some WSGI-network glue, like uwsgi or eventlet.wsgi or other. But then came 12factor and modern expectation is application is responsible for listening HTTP. Which would be great if we eliminated needless WSGI layer, but it stuck for unknown reasons. And this question well shows there is no clarity where it should be.
